### PR TITLE
st2 [1891][ADD] stock_quantity_history_date

### PR DESCRIPTION
**Pull request by TomohiroAbe-Quartile**
[1891](https://www.quartile.co/web#id=1891&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

File: inventory_valuate_date_hls/wizards/stock_quantity_history.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+class StockQuantityHistory(models.TransientModel):
+    _inherit = "stock.quantity.history"
+
+    date_range = fields.Many2one("date.range", string="Date Range")
+    inventory_valuate_date = fields.Date(
+        string="Inventory Valuate Date",
+        related="date_range.inventory_valuate_date",
+        readonly=True,
+        store=True,
+    )
```
Comment (TomohiroAbe-Quartile) [inline inventory_valuate_date_hls/wizards/stock_quantity_history.py]: @yostashiro @tack-yabeQuartile @kakurai8 
Sorry, I added this code but couldn't add field.
Please give me some advise.

File: inventory_valuate_date_hls/wizards/stock_quantity_history.py [hunk 2]
```diff
@@ -4,10 +4,4 @@
...
     _inherit = "stock.quantity.history"
 
-    date_range = fields.Many2one("date.range", string="Date Range")
-    inventory_valuate_date = fields.Date(
-        string="Inventory Valuate Date",
-        related="date_range.inventory_valuate_date",
-        readonly=True,
-        store=True,
-    )
+    inventory_valuate_month = fields.Many2one("date.range", readonly=True,)
```
Comment (yostashiro) [inline inventory_valuate_date_hls/wizards/stock_quantity_history.py]: I'd suggest `valuation_month_id` for the field name following the convention (i.e. the field name should end with `_id` for a many2one field).

Why readonly?

**Review by yostashiro — COMMENTED** [review]
Please change the module name to `stock_quantity_history_propose_date`.

**Review by yostashiro — COMMENTED** [review]
Please change the module name to stock_quantity_history_date or something in that line.

File: stock_quantity_history_date/models/date_range.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2021 Quartile.co
+#   (http://www.quartile.co)
```
Comment (yostashiro) [inline stock_quantity_history_date/models/date_range.py]:
```suggestion
# Copyright 2021 Quartile Limited
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```

File: stock_quantity_history_date/wizards/stock_quantity_history.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2021 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
```
Comment (yostashiro) [inline stock_quantity_history_date/wizards/stock_quantity_history.py]:
```suggestion
# Copyright 2021 Quartile Limited
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```

File: stock_quantity_history_date/wizards/stock_quantity_history.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+from odoo import api, fields, models
+
+
+class StockQuantityHistory(models.TransientModel):
+    _inherit = "stock.quantity.history"
+
+    date_range_id = fields.Many2one("date.range", string="Date Range")
+    valuate_date = fields.Datetime(
+        string="Inventory Valuate Date", compute="_compute_valuate_date"
+    )
```
Comment (yostashiro) [inline stock_quantity_history_date/wizards/stock_quantity_history.py]:
```suggestion
```

File: stock_quantity_history_date/wizards/stock_quantity_history.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    @api.onchange("date_range_id")
+    def _compute_valuate_date(self):
+        for stock in self:
+            if stock.date_range_id.inventory_valuate_date:
+                stock.valuate_date = stock.date_range_id.inventory_valuate_date
```
Comment (yostashiro) [inline stock_quantity_history_date/wizards/stock_quantity_history.py]:
```suggestion
    def _onchange_date_range_id(self):
        for rec in self:
            if rec.date_range_id:
                rec.date = rec.date_range_id.inventory_valuate_date
```

File: stock_quantity_history_date/wizards/stock_quantity_view.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_stock_account_quantity_history_location" model="ir.ui.view">
+        <field name="name">Valuation Report</field>
+        <field name="model">stock.quantity.history</field>
+        <field name="inherit_id" ref="stock_account.view_stock_quantity_history"/>
+        <field name="arch" type="xml">
+            <field name="compute_at_date" position="after">
+                <field name="date_range_id"/>
+                <field name="valuate_date" attrs="{'invisible': [('date_range_id', '=', False)]}"/>
```
Comment (yostashiro) [inline stock_quantity_history_date/wizards/stock_quantity_view.xml]:
```suggestion
```

File: stock_quantity_history_date/wizards/stock_quantity_view.xml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_stock_account_quantity_history_location" model="ir.ui.view">
+        <field name="name">Valuation Report</field>
+        <field name="model">stock.quantity.history</field>
+        <field name="inherit_id" ref="stock_account.view_stock_quantity_history"/>
```
Comment (yostashiro) [inline stock_quantity_history_date/wizards/stock_quantity_view.xml]: It's a random reference. Inherit the view from the stock module.

File: stock_quantity_history_date/readme/DESCRIPTION.rst [hunk 9]
```diff
@@ -0,0 +1,3 @@
+This module does following:
+- Adds inventory_valuate_date field in date_range module.
+- Shows inventory_valuate_date in Inventory Valuation wizard.
```
Comment (yostashiro) [inline stock_quantity_history_date/readme/DESCRIPTION.rst]:
```suggestion
This module does the following:

- Adds the valuation_date field in the date.range model.
- Adds the date_range_id field in the Inventory Valuation wizard to propose the valudation date from the selected date range.
```

File: stock_quantity_history_date/models/date_range.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+
+from odoo import fields, models
+
+
+class DateRange(models.Model):
+    _inherit = "date.range"
+
+    inventory_valuate_date = fields.Datetime(string="Inventory Valuate Date")
```
Comment (yostashiro) [inline stock_quantity_history_date/models/date_range.py]:
```suggestion
    valuation_date = fields.Datetime(
        string="Inventory Valuate Date",
        help="Set the date that is intended to be used for showing the inventory at date."
    )
```
Translation of the help: 時点在庫出力で使用すべき日時を設定してください。

File: stock_quantity_history_date/i18n/ja_JP.po [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+#. module: stock_quantity_history_date
+#: model:ir.model.fields,field_description:stock_quantity_history_date.field_date_range__inventory_valuate_date
+#: model:ir.model.fields,field_description:stock_quantity_history_date.field_stock_quantity_history__valuate_date
+msgid "Inventory Valuate Date"
+msgstr "在庫評価日"
+
+#. module: stock_quantity_history_date
+#: model:ir.model,name:stock_quantity_history_date.model_stock_quantity_history
+msgid "Stock Quantity History"
+msgstr "在庫記録日"
```
Comment (yostashiro) [inline stock_quantity_history_date/i18n/ja_JP.po]:
```suggestion
msgstr "在庫数量履歴"
```
The translation is strange.  Please fix the translation in Transifex if it comes from the Odoo standard.

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: stock_quantity_history_date/i18n/ja_JP.po (hunk 11 above)
Comment (TomohiroAbe-Quartile) [reply stock_quantity_history_date/i18n/ja_JP.po]: Sorry, It was my translation.

File: stock_quantity_history_date/models/date_range.py (hunk 10 above)
Comment (TomohiroAbe-Quartile) [reply stock_quantity_history_date/models/date_range.py]: Thank you. I added help and translation.

File: stock_quantity_history_date/readme/DESCRIPTION.rst (hunk 9 above)
Comment (TomohiroAbe-Quartile) [reply stock_quantity_history_date/readme/DESCRIPTION.rst]: Thank you, I edit.

File: stock_quantity_history_date/wizards/stock_quantity_view.xml (hunk 8 above)
Comment (TomohiroAbe-Quartile) [reply stock_quantity_history_date/wizards/stock_quantity_view.xml]: Thank you. I confrimed work.

File: stock_quantity_history_date/wizards/stock_quantity_history.py (hunk 6 above)
Comment (TomohiroAbe-Quartile) [reply stock_quantity_history_date/wizards/stock_quantity_history.py]: Thank you.
I'd misunderstanding. Now worked as expected.

File: stock_quantity_history_date/wizards/stock_quantity_history.py (hunk 4 above)
Comment (TomohiroAbe-Quartile) [reply stock_quantity_history_date/wizards/stock_quantity_history.py]: Thank you, I edited.

File: stock_quantity_history_date/models/date_range.py [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class DateRange(models.Model):
+    _inherit = "date.range"
+
+    valuation_date = fields.Datetime(
+        string="Inventory Valuate Date",
```
Comment (yostashiro) [inline stock_quantity_history_date/models/date_range.py]:
```suggestion
```

File: stock_quantity_history_date/i18n/ja_JP.po [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+#. module: stock_quantity_history_date
+#: model:ir.actions.act_window,name:stock_quantity_history_date.date_range_action
+#: model:ir.ui.menu,name:stock_quantity_history_date.menu_valuation
+msgid "Date Ranges"
+msgstr "日付範囲"
+
+#. module: stock_quantity_history_date
+#: model:ir.model.fields,field_description:stock_quantity_history_date.field_date_range__valuation_date
+msgid "Inventory Valuate Date"
+msgstr "在庫評価日"
```
Comment (yostashiro) [inline stock_quantity_history_date/i18n/ja_JP.po]:
```suggestion
msgid "Valuation Date"
msgstr "評価日"
```

**Review by yostashiro — APPROVED** [review]
Please go ahead with the merge upon following up on the commented points.

**Pull request merged**